crates/ra_syntax/src/parser_impl/input.rs

```diff
@@ -4,11 +4,26 @@ use std::ops::{Add, AddAssign};
 
 pub(crate) struct ParserInput<'t> {
     text: &'t str,
+    /// start position of each token(expect whitespace and comment)
+    /// ```rust
+    ///  struct Foo;
+    /// ^------^---
+    /// |      |
+    /// 0      7
```

Comment:
@matklad for tailing token `;`, what's its start_offset?

Reply:
so, for `struct Foo;` I think we'll have 3 tokens and 3 start_offsets: `0, 7, 10`. It's better to double check me by just printing an example: I've touched  this code long time ago, might be forgetting the details.